be/src/storage/vectorized/schema_change.cpp

```diff
@@ -336,7 +336,7 @@ bool ChunkChanger::change_chunk(ChunkPtr& base_chunk, ChunkPtr& new_chunk, Table
 
             if (new_type == ref_type && (!is_decimalv3_field_type(new_type) ||
                                          (reftype_precision == newtype_precision && reftype_scale == newtype_scale))) {
-                if (new_type == OLAP_FIELD_TYPE_CHAR) {
+                if (new_type == OLAP_FIELD_TYPE_CHAR || (new_col->is_nullable() ^ base_col->is_nullable())) {
```

Comment:
I think (new_col->is_nullable() != base_col->is_nullable()) is more straight.

Reply:
> I think (new_col->is_nullable() != base_col->is_nullable()) is more straight.

Seems you are right, fix in https://github.com/StarRocks/starrocks/pull/1297/commits/b5e98d83d5c855a761b1f95fffcafcccf23301c9